lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -542,11 +542,12 @@ sub _generate_jobs {
             }
 
             # add properties from dedicated database columns to settings
-            $settings{TEST}              = $job_template->name || $job_template->test_suite->name;
-            $settings{MACHINE}           = $job_template->machine->name;
-            $settings{BACKEND}           = $job_template->machine->backend;
-            $settings{JOB_TEMPLATE_NAME} = $job_template->name if $job_template->name;
-            $settings{TEST_SUITE_NAME}   = $job_template->test_suite->name;
+            $settings{TEST}                   = $job_template->name || $job_template->test_suite->name;
+            $settings{MACHINE}                = $job_template->machine->name;
+            $settings{BACKEND}                = $job_template->machine->backend;
+            $settings{JOB_TEMPLATE_NAME}      = $job_template->name if $job_template->name;
+            $settings{TEST_SUITE_NAME}        = $job_template->test_suite->name;
+            $settings{TEST_SUITE_DESCRIPTION} = $job_template->description if length $job_template->description;
```

Comment:
Looks like more and more we are (ab-)using the settings table as a generic key-value for each job. I don't mind that as it is even what I originally intended but maybe opinions differ here. Voices of others?

But I wonder if we should really duplicate the description from the job template into the job settings. We do the same for other things from the job template so it seems only consequential. Maybe we should just deep-copy the job template completely into the job?

Reply:
What else would we add here besides the description? My first thought is, most of the template wouldn't be relevant, and checking for individual keys is more convenient, too.

Reply:
For what it's worth I'm happy with storing the description here, although I'd reconsider the name. It's not the test description, it's rather the job (template) description.

Reply:
Wouldn't it make sense to assign the test suite description here when no description is defined on job template level? Then we could get rid of the test suite lookup in `sub prepare_job_results` completely. I think this would be solved most elegantly by overriding `OpenQA::Schema::Result::JobTemplates::description`.

But of course this approach means duplicating a lot of data. Hard to estimate how much the database size will increase in production and whether this is a problem. We could write a migration to find out. If we decide to follow the idea mentioned in the previous paragraph we needed a migration anyways.